tests/rkt_tests.go

```diff
@@ -515,3 +557,204 @@ func checkAppStatus(t *testing.T, ctx *rktRunCtx, multiApps bool, appName, expec
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	for _, dir := range dirs {
+		podDir := path.Join(podsDir, dir.Name(), podID)
+		if _, err := os.Stat(podDir); err == nil {
+			return podDir
+		}
+	}
+	t.Fatalf("Failed to find pod directory for pod %q", podID)
+	return ""
+}
+
+// getPodInfo returns the pod info for the given pod ID.
+func getPodInfo(t *testing.T, ctx *rktRunCtx, podID string) *podInfo {
+	p := &podInfo{
+		id:       podID,
+		apps:     make(map[string]*appInfo),
+		networks: make(map[string]*networkInfo),
+	}
+
+	// Read pod manifest.
+	podManifestPath := path.Join(getPodDir(t, ctx, podID), "pod")
+	f, err := os.Open(podManifestPath)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	p.manifest, err = ioutil.ReadAll(f)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	// Fill app infos.
+	var manifest schema.PodManifest
+	if err := json.Unmarshal(p.manifest, &manifest); err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	for _, app := range manifest.Apps {
+		appName := app.Name.String()
+		p.apps[appName] = &appInfo{
+			name: appName,
+			// TODO(yifan): Get the image's name.
+			image: &imageInfo{id: app.Image.ID.String()},
+		}
+	}
+
+	// Fill other infos.
+	output, err := exec.Command("/bin/bash", "-c", fmt.Sprintf("%s status %s", ctx.cmd(), podID)).CombinedOutput()
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	parsePodInfoOutput(t, string(output), p)
+
+	return p
+}
+
+// parseImageInfoOutput parses the 'rkt image list' result into imageInfo struct.
+// For example, the 'result' can be:
+// 'sha512-e9b77714dbbfda12cb9e136318b103a6f0ce082004d09d0224a620d2bbf38133 nginx:latest 2015-10-16 17:42:57.741 -0700 PDT true'
+func parseImageInfoOutput(t *testing.T, result string) *imageInfo {
+	fields := strings.Split(result, "\t")
+	if len(fields) != 4 {
+		t.Fatalf("Failed to parse result: %q", result)
+	}
+	nameVersion := strings.Split(fields[1], ":")
+	if len(nameVersion) != 2 {
+		t.Fatalf("Failed to parse name version string: %q", fields[1])
+	}
+	importTime, err := time.Parse(defaultTimeLayout, fields[2])
+	if err != nil {
+		t.Fatalf("Failed to parse time string: %q", fields[2])
+	}
+
+	return &imageInfo{
+		id:         fields[0],
+		name:       nameVersion[0],
+		version:    nameVersion[1],
+		importTime: importTime.Unix(),
+	}
+}
+
+// getImageInfo returns the image info for the given image ID.
+func getImageInfo(t *testing.T, ctx *rktRunCtx, imageID string) *imageInfo {
+	output, err := exec.Command("/bin/bash", "-c", fmt.Sprintf("%s image list --full | grep %s", ctx.cmd(), imageID)).CombinedOutput()
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	imgInfo := parseImageInfoOutput(t, string(output))
+
+	// Get manifest
+	output, err = exec.Command("/bin/bash", "-c", fmt.Sprintf("%s image cat-manifest %s", ctx.cmd(), imageID)).CombinedOutput()
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	imgInfo.manifest = bytes.TrimSpace(output)
```

Comment:
@alban The outputs contain a new line at the end:
https://github.com/coreos/rkt/blob/master/rkt/image_cat_manifest.go#L75
https://github.com/coreos/rkt/blob/master/rkt/rkt.go#L134